### DISCUSSION TITLE
Max number of fields in an index

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#14890] [General]
Hi,

I am generating dynamic fields for an index, 5 per commodity our system manages, eventually we'll probably support 100's of commodities, so could end up with 1000s of fields, is there a hard or recommended limit to field count on an index? I've seen Elastic Search recommend no more than 1000 but not sure if this is due to their internals rather than something specific to Lucene.

Thanks

**Comment by ayende** [comment]
There is not a hard limit, and a few thousands is acceptable.
It would be expensive to _sort_ on those, mind, if you are sorting by many dynamic fields (we need to create a internal array sized to the total number of documents for each sorted field)

**Comment by nickchampion** [comment]
Thanks @ayende we wont be sorting on those fields
